### PR TITLE
[LibXDiff] Add port

### PR DESCRIPTION
**Pull request by comedinha**
**Describe the pull request**

- Which triplets are supported/not supported? Have you updated the CI baseline?
  tested triplets:
  - x64-windows
  - x64-windows-static
  - x86-windows
  - x86-windows-static
  - x64-linux (WSL)

  not tested:
  - arm-uwp
  - x64-uwp

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/libxdiff/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: libxdiff
+Version: 0.23
+Homepage: https://github.com/Drako/libxdiff
+Description: The LibXDiff library implements basic and yet complete functionalities to create file differences/patches to both binary and text files. The library uses memory files as file abstraction to achieve both performance and portability.
```
Comment (JackBoosY) [inline ports/libxdiff/CONTROL:4]:
```suggestion
Description: The LibXDiff library implements basic and yet complete functionalities to create file differences/patches to both binary and text files. The library uses memory files as file abstraction to achieve both performance and portability.
Supports: !uwp
```

File: ports/libxdiff/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
+vcpkg_fail_port_install(MESSAGE "${PORT} currently only supports being built for desktop" ON_TARGET "UWP")
+
+set(LIBXDIFF_REF 77e30f3190685efd87cce2c9c5d688cbaa1b0134)
+set(LIBXDIFF_SHA512 c559b575e6d6f06f3b3064f3e077a15d8f57422340199215a4cbd7beab527bc250347c8779a8d6f8c4e41799a032431e83c7336f86569527ab754444455b8c87)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" KEYSTONE_BUILD_STATIC)
```
Comment (JackBoosY) [inline ports/libxdiff/portfile.cmake:6]:
```suggestion
```

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -983,6 +983,8 @@ libxslt:x64-uwp=fail
 libyuv:arm-uwp=fail
 libyuv:x64-uwp=fail
 libzippp:x64-linux=ignore
+libxdiff:arm-uwp=fail
```
Comment (JackBoosY) [inline scripts/ci.baseline.txt:986]:
```suggestion
```

File: scripts/ci.baseline.txt [hunk 4]
```diff
@@ -983,6 +983,8 @@ libxslt:x64-uwp=fail
 libyuv:arm-uwp=fail
 libyuv:x64-uwp=fail
 libzippp:x64-linux=ignore
+libxdiff:arm-uwp=fail
+libxdiff:x64-uwp=fail
```
Comment (JackBoosY) [inline scripts/ci.baseline.txt:987]:
```suggestion
```

File: ports/libxdiff/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    OPTIONS -DBUILD_SHARED=${KEYSTONE_BUILD_SHARED}
+)
+
+vcpkg_install_cmake()
+
+if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL debug)
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+    if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/xdiff.dll")
```
Comment (JackBoosY) [inline ports/libxdiff/portfile.cmake:28]: Is this dynamic library installed in _${CURRENT_PACKAGES_DIR}/bin_?

File: ports/libxdiff/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    endif()
+endif()
+
+
+if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL release)    
+    if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/xdiff.dll")
```
Comment (JackBoosY) [inline ports/libxdiff/portfile.cmake:35]: Same as above.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libxdiff/portfile.cmake (hunk 5 above)
Comment (comedinha) [reply ports/libxdiff/portfile.cmake:28]: Added again.

File: ports/libxdiff/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+set(LIBXDIFF_REF 77e30f3190685efd87cce2c9c5d688cbaa1b0134)
+set(LIBXDIFF_SHA512 c559b575e6d6f06f3b3064f3e077a15d8f57422340199215a4cbd7beab527bc250347c8779a8d6f8c4e41799a032431e83c7336f86569527ab754444455b8c87)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" KEYSTONE_BUILD_SHARED)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Drako/libxdiff
+    REF ${LIBXDIFF_REF}
+    SHA512 ${LIBXDIFF_SHA512}
```
Comment (JackBoosY) [inline ports/libxdiff/portfile.cmake:12]:
```suggestion
    SHA512 ${LIBXDIFF_SHA512}
    HEAD_REF master
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**